src/services/refactors/convertToEs6Module.ts

```diff
@@ -0,0 +1,567 @@
+            ];
+            changes.replaceNodeWithNodes(sourceFile, statement, newNodes, { nodeSeparator: "\n", useNonAdjustedEndPosition: true });
+        }
+        else {
+            changes.replaceNode(sourceFile, statement, convertExportsDotXEquals(text, right), { useNonAdjustedEndPosition: true });
+        }
+    }
+
+    function convertModuleExportsToExportDefault(exported: Expression): [ReadonlyArray<Statement>, ModuleExportsChanged] {
+        const modifiers = [createToken(SyntaxKind.ExportKeyword), createToken(SyntaxKind.DefaultKeyword)];
+        switch (exported.kind) {
+            case SyntaxKind.FunctionExpression:
+            case SyntaxKind.ArrowFunction: {
+                // `module.exports = function f() {}` --> `export default function f() {}`
+                const fn = exported as FunctionExpression | ArrowFunction;
+                return [[functionExpressionToDeclaration(fn.name && fn.name.text, modifiers, fn)], true];
+            }
+            case SyntaxKind.ClassExpression: {
+                // `module.exports = class C {}` --> `export default class C {}`
+                const cls = exported as ClassExpression;
+                return [[classExpressionToDeclaration(cls.name && cls.name.text, modifiers, cls)], true];
+            }
+            case SyntaxKind.CallExpression:
+                if (isRequireCall(exported, /*checkArguementIsStringLiteral*/ true)) {
+                    // `module.exports = require("x");` ==> `export * from "x"; export { default } from "x";`
+                    const moduleSpecifier = exported.arguments[0].text;
+                    const newNodes = [
+                        makeExportDeclaration(/*exportClause*/ undefined, moduleSpecifier),
+                        makeExportDeclaration([createExportSpecifier(/*propertyName*/ undefined, "default")], moduleSpecifier),
+                    ];
+                    return [newNodes, false];
+                }
+                // falls through
+            default:
+                // `module.exports = 0;` --> `export default 0;`
+                return [[createExportAssignment(/*decorators*/ undefined, /*modifiers*/ undefined, /*isExportEquals*/ false, exported)], true];
+        }
+    }
+
+    function convertExportsDotXEquals(name: string | undefined, exported: Expression): Statement {
+        const modifiers = [createToken(SyntaxKind.ExportKeyword)];
+        switch (exported.kind) {
+            case SyntaxKind.FunctionExpression:
+            case SyntaxKind.ArrowFunction:
+                // `exports.f = function() {}` --> `export function f() {}`
+                return functionExpressionToDeclaration(name, modifiers, exported as FunctionExpression | ArrowFunction);
+            case SyntaxKind.ClassExpression:
+                // `exports.C = class {}` --> `export class C {}`
+                return classExpressionToDeclaration(name, modifiers, exported as ClassExpression);
+            default:

```

Comment:
Does this cover a reasonably short list of kinds?  It would be nice if we could have the compiler notify us when a new case requires coverage.

Reply:
Unfortunately `Expression` is a huge list of kinds. But only function/class expressions can be converted to declarations, all others should be left as expressions.